src/linelist.jl

```diff
@@ -1,132 +1,229 @@
+            if length(code) <= 2
+                return Formula(parse(Int,code))
+            elseif length(code) <= 4
+                if length(code) == 3  
+                    code = "0"*code
+                end
+                return Formula(parse(Int, code[1:2]), parse(Int, code[3:4]))
+            else
+                throw(ArgumentError("numeric codes for molecules with more than 4 chars are not "*
+                                    "supported"))
+            end
+        end
+        #otherwise, code should be "OH", "FeH", "Li", "C2", etc.
+        inds = filter(1:length(code)) do i
+            isdigit(code[i]) || isuppercase(code[i])
         end
-        a1 = atomic_symbols[parse(Int, tok[1:2])] 
-        a2 = atomic_symbols[parse(Int, tok[3:4])]
-        if a1 == a2
-            a2 = "2"
+        push!(inds, length(code)+1)
+        subcode = map(1:(length(inds)-1)) do j
+            code[inds[j]:inds[j+1]-1]
         end
-        a1 * a2
+
+        atoms = String[]
+        for s in subcode
+            num = tryparse(Int, s)
+            if num isa Int
+                previous = atoms[end]
+                for _ in 1:(num-1)
+                    push!(atoms, previous)
+                end
+            else
+                push!(atoms, s)
+            end
+        end
+
+        sort!(atoms, by=s->atomic_numbers[s])
+        new(String.(atoms))
     end
 
-    atom_or_molecule * ionization
+    """
+        Formula(symbols::Int...)
+
+    Create a Formula by providing the atomic symbols of its components as arguments
+    """
+    function Formula(symbols::AbstractString...)
+        new(sort(collect(String.(symbols)), by=s->atomic_numbers[s]))
+    end
+
+    function Formula(Z::Int...)
+        new([atomic_symbols[z] for z in sort(collect(Z))])
+    end
+
+    function Formula(symbols::Vector{AbstractString})
+        @assert issorted(symbols, by=s->atomic_numbers[s])
+        new(String.(symbols))
+    end
+end
+
+#pretty-print lines in REPL and jupyter notebooks
+function Base.show(io::IO, m::MIME"text/plain", b::Formula)
+    print(io, *(b.atoms...))
+end
+
+function (==)(b1::Formula, b2::Formula)
+    b1.atoms == b2.atoms
+end
+
+function hash(b::Formula, h::UInt)
+    hash(b.atoms, h)
 end
 
 """
-    strip_ionization(code)
+    ismolecule(b::Formula)
 
-get the chemical symbol for the element of the species
+`true` when `b` is composed of more than one atom
 """
-strip_ionization(code::AbstractString)::String = split(code, '_')[1]
+ismolecule(b::Formula) = length(b.atoms) > 1
 
 """
-    ismolecule(species)
+    mass(b::Formula)
 
-true if the string passed represents a molecule (with or without its ionization state)
+Returns the mass [g] of `b`.
 """
-function ismolecule(species)
-    count = 0
-    for c in species
-        if c == '_'
-            break
-        else
-            count += isdigit(c) + isuppercase(c)
-        end
-    end
-    count > 1
+function mass(b::Formula)
```

Comment:
Yeah, it would be great if we could come with a better name for this.

Reply:
Thoughts on `get_mass`?  I don't love it but I can't think of anything better?

Reply:
Yeah, that or `particle_mass` would work. 

Alternatively, I think the short name would probably be fine if we explicitly placed this in some submodule (maybe a `Species` submodule?)

Reply:
> Alternatively, I think the short name would probably be fine if we explicitly placed this in some submodule (maybe a Species submodule?)

True!  I guess that's kinda what they are for...